File: BitManipulation/1356_Sort_Integers_by_The_Number_of_1_Bits/1356_Sort_Integers_by_The_Number_of_1_Bits.cpp
class Solution {
public:
    vector<int> sortByBits(vector<int>& arr) {
        auto compare = [](int a, int b) 
        {
            int ca = bitset<32>(a).count();
            int cb = bitset<32>(b).count();
            if(ca == cb)
                return a < b;
            return ca < cb;
        };

        sort(arr.begin(), arr.end(), compare);
        return arr; 
    }
};

File: BitManipulation/1356_Sort_Integers_by_The_Number_of_1_Bits/1356_Sort_Integers_by_The_Number_of_1_Bits.java
class Solution {
    public int[] sortByBits(int[] arr) {
        // Convert the array to Integer array for custom sorting
        Integer[] integerArr = Arrays.stream(arr).boxed().toArray(Integer[]::new);

        // Sort using custom comparator
        Arrays.sort(integerArr, Comparator.comparingInt((Integer i) -> Integer.bitCount(i))
                                          .thenComparingInt(i -> i));

        // Convert back to int array
        return Arrays.stream(integerArr).mapToInt(Integer::intValue).toArray();
    }
}